src/Psalm/Config.php

```diff
@@ -1817,7 +1817,7 @@ private function getPluginClassForPath(Codebase $codebase, string $path, string
     public function shortenFileName(string $to): string
     {
         if (!is_file($to)) {
-            return preg_replace('/^' . preg_quote($this->base_dir, '/') . '/', '', $to, 1);
+            return preg_replace('/^' . preg_quote($this->base_dir . DIRECTORY_SEPARATOR, '/') . '?/', '', $to, 1);
```

Comment:
I think directory separator shouldn't be optional in the regexp:
![image](https://github.com/vimeo/psalm/assets/57403/f3b3ea41-1787-4380-b956-735412f3035f)